src/proof_common.h

```diff
@@ -12,6 +12,7 @@ std::vector<unsigned char> ConvertIntegerToBytes(integer x, uint64_t num_bytes)
     }
     for (int iter = 0; iter < num_bytes; iter++) {
         auto byte = (x % integer(256)).to_vector();
+        if (byte.empty()) byte.push_back(0);
```

Comment:
this whole function seems extremely inefficient. It could just be a single call to `mpz_export()` as far as I can tell. but that's for another PR